src/lib/Microsoft.Health.Fhir.Ingest.Template/CalculatedFunction/JMESPathExpressionEvaluator.cs

```diff
@@ -0,0 +1,50 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Collections.Generic;
+using DevLab.JmesPath;
+using EnsureThat;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Health.Fhir.Ingest.Template.CalculatedFunction
+{
+    public class JMESPathExpressionEvaluator : IExpressionEvaluator
+    {
+        private JmesPath.Expression _jmespathExpression;
+        private Expression _expression;
+
+        public JMESPathExpressionEvaluator(
+            JmesPath.Expression jmespathExpression,
+            Expression templateExpression)
+        {
+            _jmespathExpression = EnsureArg.IsNotNull(jmespathExpression, nameof(jmespathExpression));
+            _expression = EnsureArg.IsNotNull(templateExpression, nameof(templateExpression));
+        }
+
+        public JToken SelectToken(JToken data)
+        {
+            var result = _jmespathExpression.Transform(data).AsJToken();
```

Comment:
The call to transform always returns a `JmesPathArgument` and doesn't throw. Same thing for `AsJToken`.  As it turns out, if there is no match it will return a `JToken` with a `JTokenType` of NULL. So I'll update the code to account for that.